### PR TITLE
Milestone: Members-only - Authentication with Ruby on Rails

### PR DESCRIPTION
**Pull request by alvarorf**
screenshot
In this pull request, we add the necessary code for the fourth project in the Ruby on Rails curriculum. The main purpose of this project was to get to know how the authentication system works in Rails, specifically using the _devise_  gem. Users and posts can be created and edited, but only authorized (signed in) users can create new posts. Also, only signed-in users can see the post's authors(a more thorough description may be found in the Readme).

Thank you for any errors that you may find and for any suggestions that you may have.

### CODE REVIEW

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello Team!, Great job! 👏  
Both of you have done well on this project. Excellent PR description and commits. Your readme looks professional. The screenshot of the app will add color to it.

Let's quickly add some finishing touches and request another review.

- [x] In your readme file, add `yarn add` instructions just after `bundle install`, as I could not proceed with `rails db:migrate` without update the yarn packages.

- [x] Kindly update your signup instructions for the password field. Instructions say a minimum of 6 characters, but your validations for the password is 8.

- [x] I encountered an error while creating a post: 
Screen Shot 2020-11-13 at 6 23 13 PM


* [x] There should be absolutely NO logic in views. >Views are only for displaying information that becomes available to it from its corresponding controller not generating it. **Resources**
* [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
* [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
*[Rails view helpers best practices guide](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4)
* [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    <p class="navbar-text float-right">
+        <% if user_signed_in? %>
+        Logged in as <strong><%= current_user.email %></strong>.
+        <%= link_to 'Edit profile', edit_user_registration_path, :class => 'navbar-link' %> |
+        <%= link_to "Logout", destroy_user_session_path, method: :delete, :class => 'navbar-link'  %>
+        <% else %>
+        <%= link_to "Sign up", new_user_registration_path, :class => 'navbar-link'  %> |
+        <%= link_to "Login", new_user_session_path, :class => 'navbar-link'  %>
+        <% end %>
```
[the remark this answers is not in the document]
Comment (alvarorf) [reply app/views/layouts/application.html.erb]: Thanks, you are completely right. We removed this in the next commits.

File: app/views/posts/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<%= form_with(model: post, local: true) do |form| %>
+  <% if post.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(post.errors.count, "error") %> prohibited this post from being saved:</h2>
```
[the remark this answers is not in the document]
Comment (alvarorf) [reply app/views/posts/_form.html.erb:4]: Thanks, you are completely right. We removed this in the next commits.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    </tr>
+  </thead>
+
+  <tbody>
+    <% @posts.each do |post| %>
+      <tr>
+      <div>
+      <% if user_signed_in? %>
+        <td><%= post.user.username %></td>
+      <% end %>
```
[the remark this answers is not in the document]
Comment (alvarorf) [reply app/views/posts/index.html.erb]: Thanks, you are completely right. We removed this in the next commits.

**Review by rOluochKe — APPROVED** [review]
Hello Team
### You got it right :clap: The project has been approved. Congratulations :boom: :100:
### Merge and submit the project. Good luck with the next one.
- Happy coding!

**Pull request merged**